static/js/map.js

```diff
@@ -997,82 +997,85 @@ function gymLabel(gym, includeMembers = true) {
-                <span class='pokestop navigate'><a href='javascript:void(0);' onclick='javascript:openMapDirections(${latitude},${longitude});' title='Open in Google Maps'; class='pokestop lure'>${latitude.toFixed(6)}, ${longitude.toFixed(7)}</a></span>
-              </div>
-            </div>
-          </div>`
-    } else {
-        str = `
-            <div>
-              <div class='pokestop nolure'>
-                Pokéstop
-              </div>
-              <div>
-                <img class='pokestop sprite' src='static/images/pokestop//Pokestop.png'>
-              </div>
-              <div>
-                ${questtext}
-              </div>
-              <div>
-                <span class='pokestop navigate'><a href='javascript:void(0);' onclick='javascript:openMapDirections(${latitude},${longitude});' title='Open in Google Maps'; class='pokestop nolure'>${latitude.toFixed(6)}, ${longitude.toFixed(7)}</a></span>
-              </div>
+        pokestopExpiration = `
+          <div class='pokestop-expire'>
+            <span class='label-countdown' disappears-at='${expireTime}'>00m00s</span> left (${moment(expireTime).format('HH:mm')})
+          </div>
+        `
+    }
+
+    if (typeof pokestop.image !== 'undefined' && pokestop.image !== null) {
+        pokestopImg = `<img class='pokestop imgcircle ${luredClass}' src='${pokestop.image}'>`
+    }
+
+    const quest = pokestop.quest_raw
+    if (quest['is_quest']) {
+        let image = ''
+        let rewardText = ''
+        let width = 40
+
+        switch (quest['quest_reward_type_raw']) {
+            case '2':
+                image = 'static/quest/reward_' + quest['item_id'] + '_1.png'
+                rewardText = quest['item_amount'] + ' ' + i8ln(quest['item_type'])
+                width = 40
+                break
+            case '3':
+                image = 'static/quest/reward_stardust.png'
+                rewardText = quest['item_amount'] + ' ' + i8ln(quest['item_type'])
+                width = 40
+                break
+            case '7':
+                const pokemonId = quest['pokemon_id']
+                if (generateImages) {
+                    image = `pkm_img?pkm=${pokemonId}`
+                } else {
+                    image = pokemonSprites(quest['pokemon_id']).filename
+                }
+                rewardText = quest['quest_pokemon_name']
+                width = 40
+                break
+        }
+
+        questText = `
+          <div class="pokestop-quest">
+            <div class="pokestop-quest__reward">
+              <div class="pokestop-quest__reward-image"><img src="${image}" width="${width}" /></div>
+              <div class="pokestop-quest__reward-text">${rewardText}</div>
             </div>
+            <div class="pokestop-quest__task">${quest['quest_task']}</div>
           </div>`
     }
 
-    return str
+    return `
+      <div>
+        <div class='pokestop ${luredClass}'>
+          ${pokestop.name}
```

Comment:
Missing fallback to "pokestop" for any stop that has not been scanned (new stops, or stops in areas not quest scanned)

Reply:
Thanks! How about now?

Reply:
![image](https://user-images.githubusercontent.com/757637/53685512-ebbdb300-3d1b-11e9-80dd-88c78afbcf39.png)
We fallback to the old layout when the stop does not have a name nor an image. So I think we're good to go and I will merge the PR